### PR TITLE
Add rust injections for c, c++, html, js, py

### PR DESCRIPTION
**Pull request by pickfire**
cc @the-mikedavis 

Example
```rust
use inline_python::python;

fn main() {
    let who = "world";
    let n = 5;
    python! {
        import csv  # first import highlight won't work, not sure why
        for i in range('n):
            print(i, "Hello", 'who)
        print("Goodbye")
    }
}
```

### CODE REVIEW

File: runtime/queries/rust/injections.scm [hunk 1]
```diff
@@ -1,14 +1,52 @@
...
+    name: (identifier) @_python (#eq? @_python "python"))
+  (token_tree) @injection.content)
+ (#set! injection.language "python")
+ (#set! injection.include-children))
+
+((macro_invocation
+  macro: (identifier) @_python (#eq? @_python "python")
+  (token_tree) @injection.content)
+ (#set! injection.language "python")
+ (#set! injection.include-children))
```
Comment (pickfire) [inline runtime/queries/rust/injections.scm:43]: Not sure how can I simplify this so I leave it like this for now.

**Comment by the-mikedavis** [comment]
I'm not totally sure what's happening here with the injections within `(token_tree)`. The results are same across helix `master`, `incremental`, and `tree-sitter-cli`: the injection works on individual identifiers under the token_tree, but any python that uses multiple identifiers isn't parsed correctly. For example, in the string of identifiers making up the comment after the import, the "#" is highlighted as a comment but the subsequent nodes are not highlighted as comments. Based on the range of the outermost `(token_tree)` and the description of `injection.include-children` in the [tree-sitter docs](https://github.com/tree-sitter/tree-sitter/blob/e96ee19901947d8550abbcf425791c2a411c8d97/docs/section-4-syntax-highlighting.md?plain=1#L365)

> * `injection.include-children` - indicates that the `@injection.content` node's *entire* text should be re-parsed, including the text of its child nodes. By default, child nodes' text will be *excluded* from the injected document.

I would expect that the contents of `(token_tree)` shouldn't matter, as the entire text of that node should be re-parsed.

I think what's happening here is that some of the rules from the rust grammar are excluding nodes from the token tree that are important to be parsed for python's sake. For example, it looks like the `#` token is not present under `(token_tree)`.

@archseer do you have thoughts about the behavior here?

**Comment by archseer** [comment]
I find most of these a bit niche and not sure if we want these in by default. i.e. https://github.com/search?l=Rust&q=inline_python&type=Code is barely used

**Comment by pickfire** [comment]
> I find most of these a bit niche and not sure if we want these in by default. i.e. https://github.com/search?l=Rust&q=inline_python&type=Code is barely used

Maybe have them commented out by default? But I doubt anyone using it wants to dig into the highlight to enable support for it. Will enabling it slows down the tree-sitter highlighting?

**Comment by the-mikedavis** [comment]
The injections for python in rust may work better with the new `(#set injections.include-unnamed-children true)` rule from https://github.com/helix-editor/helix/pull/3129

I haven't tried it out though.

**Comment by pickfire** [comment]
@the-mikedavis Do you think I should continue to work on this or just close it?

**Comment by the-mikedavis** [comment]
IMO it's more work than it's worth to make these queries look good but I'm biased since I haven't used Rust in a cross-language way like this before :P

**Pull request closed without merging**